src/Kraken/Config.hs

```diff
@@ -22,13 +26,60 @@ defaultKrakenConfig = KrakenConfig {
   numberOfRetries = 1
  }
 
-loadConfig :: FilePath -> IO (KrakenConfig, Config)
+load :: FilePath -> IO Object
+load p = BS.readFile p >>= return . decodeEither >>= \case
+  Right o -> return o
+  Left m -> error $ "Kraken.Config.load: decode failure ‘" ++ m ++ "’"
```

Comment:
Can we use `throwIO` here? (`error`, `undefined`, etc. don't behave as expected in certain corner cases, e.g. behavior may change depending on optimization levels.)

Generally the idea was that loading the config should be able to abort the process in case it can't construct the configuration successfully. For that reason, it has to be an `IO` operation, and use `throwIO` (or `exitWith`).

(Also possible is something like `withConfig :: (KrakenConfig -> IO a) -> IO a`.)